UC-Berkeley-Extension/global
# Exposing JSON to Views

## Asana Ticket(s)
- See ticket [Exposing JSON to Views](https://app.asana.com/0/0/1194174850852910/f).

## Description

Creates a test View to pull data directly from SPoCC.

## Testing instructions

* Checkout this branch
* From the project root, run `composer install`
* From inside the VM, run `../scripts/refresh.sh` or `../scripts/setup.sh` depending on whether you want a fresh site build.
* Visit https://global.ddev.site/spocc-test
* You should see data from https://voices.berkeley.edu/international-feed/post/all/tag/all/program/all
* Login and go to https://global.ddev.site/admin/structure/views/view/spocc_test/edit to see the test View.

## Remaining tasks

There is real potential here. A few things would remain to be done as module improvements:

* Do not escape HTML by default when printing fields.
* Handle arrays of data in fields
  * Note how we handle the `Academic Areas` with `academicAreas/0/name` will only pull the first item. There would need to be a wildcard system to pull that data as an array.

The above two items are ~ 12 hours of developer time.
